src/transformers/generation/utils.py

```diff
@@ -2764,7 +2770,11 @@ def _contrastive_search(
             # compute the degeneration penalty and re-rank the candidates based on the degeneration penalty and the
             # model confidence. Keeping `selected_idx` on CPU enables multi-device contrastive search and doesn't
             # introduce (noticeable) slowdowns on single-device runs.
-            selected_idx = _ranking_fast(context_hidden, next_hidden, top_k_probs, penalty_alpha, top_k)
+            selected_idx = _ranking_fast(
+                context_hidden, next_hidden, top_k_probs, cosine_matrix_mask, penalty_alpha, top_k
+            )
+            if not self.config.is_encoder_decoder:
```

Comment:
(this line might need to be removed, depending on the final form of the suggestion above -- i.e. if we're able to create `cosine_matrix_mask` all the times)